apps/openmw/mwstate/quicksavemanager.cpp

```diff
@@ -0,0 +1,73 @@
+#include "quicksavemanager.hpp"
+
+#include <sstream>
+
+MWState::QuickSaveManager::QuickSaveManager(std::string &saveName, int maxSaves){
+    this->saveName = saveName;
+    this->maxSaves = maxSaves;
+    this->oldestSlotVisited = NULL;
+    this->oldestSlotId = 0;
+    this->slotsVisited = 0;
+}
+
+void MWState::QuickSaveManager::visitSave(const Slot *saveSlot){
+    int slotId;
+    if(tryExtractSlotId(saveSlot->mProfile.mDescription, slotId)){
+        ++slotsVisited;
+        if(isOldestSave(saveSlot)){
+            oldestSlotVisited = saveSlot;
+            oldestSlotId = slotId;
+        }
+    }
+}
+
+bool MWState::QuickSaveManager::isOldestSave(const Slot *compare){
+    if(oldestSlotVisited == NULL)
+        return true;
+    return (compare->mTimeStamp < oldestSlotVisited->mTimeStamp);
+}
+
+bool MWState::QuickSaveManager::tryExtractSlotId(const std::string &slotName, int &extractedId){
+    std::istringstream formattedExtractor = std::istringstream(slotName);
```

Comment:
Looks like GCC 4, which CI uses, [is not happy](https://gcc.gnu.org/bugzilla/show_bug.cgi?id=54316) about istringstream.
Can you look for another solution?

Reply:
Perhaps `std::istringstream formattedExtractor(slotName);` would work instead. I don't see a reason to create a temporary that's move-constructed into the real object.